src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/items/GrapplingHook.java

```diff
@@ -19,46 +19,68 @@
-			    	Arrow arrow = (Arrow) projectile;
-			    	Bat b = (Bat) p.getWorld().spawnEntity(p.getLocation(), EntityType.BAT);
-			    	b.setCanPickupItems(false);
-			    	b.addPotionEffect(new PotionEffect(PotionEffectType.INVISIBILITY, 100000, 100000));
-			    	b.setLeashHolder(arrow);
+        return (e, p, item) -> {
+            if (SlimefunManager.isItemSimiliar(item, SlimefunItems.GRAPPLING_HOOK, true)) {
+                UUID uuid = p.getUniqueId();
+                if (e.getClickedBlock() == null && !utilities.jumpState.containsKey(uuid)) {
+                    e.setCancelled(true);
+                    if (p.getInventory().getItemInOffHand().getType() == Material.BOW) {
+                        // Cancel, to fix dupe #740
+                        return false;
+                    }
+                    utilities.jumpState.put(uuid, p.getInventory().getItemInMainHand().getType() != Material.SHEARS);
+                    if (p.getInventory().getItemInMainHand().getType() == Material.LEAD)
+                        PlayerInventory.consumeItemInHand(p);
 
-			    	utilities.damage.add(p.getUniqueId());
-			    	utilities.remove.put(p.getUniqueId(), new Entity[] {b, arrow});
-				}
-				return true;
-			}
-			else return false;
-		};
-	}
+                    Vector direction = p.getEyeLocation().getDirection().multiply(2.0);
+                    Arrow arrow = p.getWorld().spawn(p.getEyeLocation().add(direction.getX(), direction.getY(), direction.getZ()), Arrow.class);
+                    arrow.setShooter(p);
+                    arrow.setVelocity(direction);
+
+                    Bat b = (Bat) p.getWorld().spawnEntity(p.getLocation(), EntityType.BAT);
+                    b.setCanPickupItems(false);
+                    b.setAI(false);
+                    b.addPotionEffect(new PotionEffect(PotionEffectType.INVISIBILITY, 100000, 100000));
+                    b.setLeashHolder(arrow);
+
+                    utilities.damage.add(uuid);
+                    utilities.remove.put(uuid, new Entity[]{b, arrow});
+
+                    // To fix issue #253
+                    Bukkit.getScheduler().scheduleSyncDelayedTask(SlimefunPlugin.instance, () -> {
+                        if (utilities.jumpState.containsKey(uuid)) {
+                            utilities.arrows.remove(uuid);
+
+                            for (Entity n : utilities.remove.get(uuid)) {
+                                if (n.isValid()) n.remove();
+                            }
 
+                            Bukkit.getScheduler().scheduleSyncDelayedTask(SlimefunPlugin.instance, () -> {
+                                utilities.damage.remove(uuid);
+                                utilities.jumpState.remove(uuid);
+                                utilities.remove.remove(uuid);
+                            }, 20L);
+                        }
+                    }, (int) Slimefun.getItemValue("GRAPPLING_HOOK", "despawn-seconds") * 20);
```

Comment:
You should save this setting as a field instead of grabbing it from the file everytime.

Note that you need to set the field AFTER register(boolean) was called.
Easiest approach would be to override register, call super.register and then set the field

Reply:
or adding a postRegister() method to SlimefunItem?

Reply:
There already is, I think it's install() or create()

You need to look up which one it is.
They should be renamed to pre and post register though at some point.